apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderCloneModal.vue

```diff
@@ -0,0 +1,256 @@
+} from '@cloudforet/mirinae';
+
+import type { ListResponse } from '@/schema/_common/api-verbs/list';
+import { RESOURCE_GROUP } from '@/schema/_common/constant';
+import type { DashboardModel, DashboardCreateParams } from '@/schema/dashboard/_types/dashboard-type';
+import type { FolderCreateParams } from '@/schema/dashboard/_types/folder-type';
+import type { WidgetModel, WidgetListParams } from '@/schema/dashboard/_types/widget-type';
+import type { PublicFolderCreateParameters } from '@/schema/dashboard/public-folder/api-verbs/create';
+import { store } from '@/store';
+import { i18n } from '@/translations';
+
+import { useAppContextStore } from '@/store/app-context/app-context-store';
+import { useDashboardStore } from '@/store/dashboard/dashboard-store';
+import { useAllReferenceStore } from '@/store/reference/all-reference-store';
+import type { CostDataSourceReferenceMap } from '@/store/reference/cost-data-source-reference-store';
+
+import { showSuccessMessage } from '@/lib/helper/notice-alert-helper';
+
+import ErrorHandler from '@/common/composables/error/errorHandler';
+import { useProxyValue } from '@/common/composables/proxy-state';
+
+import { gray } from '@/styles/colors';
+
+import { getSharedDashboardLayouts } from '@/services/dashboards/helpers/dashboard-share-helper';
+import { useDashboardMainPageStore } from '@/services/dashboards/stores/dashboard-main-page-store';
+
+
+
+const DELETE_TABLE_FIELDS = [
+    { name: 'name', label: 'Name' },
+    { name: 'location', label: 'Location' },
+    { name: 'private', label: 'Make Private' },
+];
+interface Props {
+    visible?: boolean;
+}
+const props = withDefaults(defineProps<Props>(), {
+    visible: false,
+});
+const emit = defineEmits<{(e: 'update:visible', visible: boolean): void,
+}>();
+const appContextStore = useAppContextStore();
+const allReferenceStore = useAllReferenceStore();
+const dashboardStore = useDashboardStore();
+const dashboardMainPageStore = useDashboardMainPageStore();
+const dashboardMainPageState = dashboardMainPageStore.state;
+const dashboardMainPageGetters = dashboardMainPageStore.getters;
+const storeState = reactive({
+    isAdminMode: computed(() => appContextStore.getters.isAdminMode),
+    costDataSource: computed<CostDataSourceReferenceMap>(() => allReferenceStore.getters.costDataSource),
+});
+const state = reactive({
+    loading: false,
+    proxyVisible: useProxyValue<boolean>('visible', props, emit),
+    privateMap: {} as Record<string, boolean>,
+});
+
+/* Util */
+const getUniqueFolderName = (name: string): string => {
+    const _newName = `Clone - ${name}`;
+    return dashboardMainPageGetters.existingFolderNameList.find((n) => n === _newName) ? getUniqueFolderName(_newName) : _newName;
+};
+const getUniqueDashboardName = (name: string): string => {
+    const _newName = `Clone - ${name}`;
+    return dashboardMainPageGetters.existingDashboardNameList.find((n) => n === _newName) ? getUniqueDashboardName(_newName) : _newName;
+};
+
```

Comment:
idea: These utils would be useful if it is in util package :)

Reply:
I made it!